nanodex/utils/pipeline_state.py

```
@@ -0,0 +1,165 @@
+        self.state["updated_at"] = datetime.now().isoformat()
+        with open(self.state_file, "w") as f:
+            json.dump(self.state, f, indent=2)
+
+    def get_step_status(self, step_name: str) -> StepStatus:
+        """Get status of a specific step."""
+        if step_name in self.state["steps"]:
+            status_value = self.state["steps"][step_name]["status"]
+            return StepStatus(status_value)
+        return StepStatus.PENDING
+
+    def set_step_status(self, step_name: str, status: StepStatus, data: Dict = None):
```

Ruff (0.14.4)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)